arch/risc-v/src/common/riscv_initialize.c

```diff
@@ -105,6 +105,19 @@ void up_initialize(void)
   riscv_pminitialize();
 #endif
 
+#ifdef CONFIG_ARCH_DMA
+  /* Initialize the DMA subsystem if the weak function arm_dma_initialize has
+   * been brought into the build
+   */
+
+#ifdef CONFIG_HAVE_WEAKFUNCTIONS
+  if (riscv_dma_initialize)
```

Comment:
Maybe this can be handled more easily with macro instead of weak function?

Reply:
This is how we do this for ARM.

Reply:
Ok. I will review that part. I know that some weak functions were changed to macro recently, but do not recall the reason.